src/bindings/python/flux/cli/plugin.py

```diff
@@ -0,0 +1,180 @@
+##############################################################
+# Copyright 2024 Lawrence Livermore National Security, LLC
+# (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+#
+# This file is part of the Flux resource manager framework.
+# For details, see https://github.com/flux-framework.
+#
+# SPDX-License-Identifier: LGPL-3.0
+##############################################################
+
+import glob
+from abc import ABC
+from os import getenv
+
+from flux.conf_builtin import conf_builtin_get
+from flux.importer import import_path
+
+
+class CLIPluginOption:
+    """Wrap Argparse's add_argument method with a class
+
+    Args:
+
+        name (str): Long option string being added (must begin with ``--``).
+        prefix (str): By default, ``--ex-`` is added as a prefix to ensure
+            arguments are namespaced. However, other flux-framework projects
+            can override this (as can other projects, though it is not
+            advised).
+
+    Other keyword arguments, except ``dest=`` are passed along to
+    ``ArgumentParser.add_argument`` unchanged. ``dest=`` is a special case,
+    where dashes are replaced with underscores.
+    """
+
+    def __init__(self, name, prefix="--ex-", **kwargs):
+        if not name.startswith("--"):
+            raise ValueError("Plugins must register only long options.")
+        if prefix:
+            if "dest" not in kwargs:
+                kwargs["dest"] = name[2:].replace("-", "_")
+            self.name = name.replace("--", prefix)
+        self.__dict__.update(kwargs)
```

Comment:
Just curious what you get using `self.__dict__` vs just the more explicit use of an attribute like `self.kwargs`?

Reply:
More confusion, as is always the goal. (Fixed)